Makefile

```diff
@@ -32,18 +32,43 @@ SUB := $(addsuffix .sub, $(ALL))
 build: build.sub
 build-linux: build-linux.sub
 build-darwin: build-darwin.sub
-clean: clean.sub
 docker: docker.sub
 lint: lint.sub
 push: push.sub
 test: test.sub
 
+clean: clean.sub
+	rm -f .dockerInit
+	rm -f .buildImage
+	docker rmi -f buildImage > /dev/null 2>&1 || true
+
+# Use this target when you want to build everything using docker containers.
+# Good for cases when you don't have the tools installed (like glide or go).
+docker-all: .buildImage
+	docker run --rm -ti \
+	  -v $(PWD):/go/src/github.com/kubernetes-incubator/service-catalog \
+	  buildimage \
+	  make .dockerInit all
+
+# .dockerInit tells us if our vendor stuff if out of date or not.
+# And if so we'll run init under our docker build.  For non-docker builds
+# it is assumed you'll run "make init" manually.
+.dockerInit: glide.yaml
```

Comment:
Is there a reason for the dot at the start of the target name? Other than the fact that it's been coupled (unnecessarily; see next comment) to the existence of a `.dockerInit` file? Similar comment goes for `.buildImage` target.